der/derive/src/lib.rs

```diff
@@ -214,7 +215,44 @@ pub fn derive_choice(input: TokenStream) -> TokenStream {
 #[proc_macro_error]
 pub fn derive_enumerated(input: TokenStream) -> TokenStream {
     let input = parse_macro_input!(input as DeriveInput);
-    DeriveEnumerated::new(input).to_tokens().into()
+    DeriveEnumerated::new(input, false).to_tokens().into()
+}
+
+/// Derive decoders and encoders for ASN.1 [`Integer`] types on a
+/// C-like `enum` type. This macro is identical to [`Enumerated`] except for
+/// the ASN.1 tag.
+///
+/// # Usage
+///
+/// The `Integer` proc macro requires a C-like enum which impls `Copy`
+/// and has a `#[repr]` of `u8`, `u16`, or `u32`:
+///
+/// ```ignore
+/// use der::Integer;
+///
+/// #[derive(Integer, Copy, Clone, Debug, Eq, PartialEq)]
+/// #[repr(u32)]
+/// pub enum CrlReason {
+///     Unspecified = 0,
+///     KeyCompromise = 1,
+///     CaCompromise = 2,
+///     AffiliationChanged = 3,
+///     Superseded = 4,
+///     CessationOfOperation = 5,
+///     CertificateHold = 6,
+///     RemoveFromCrl = 8,
+///     PrivilegeWithdrawn = 9,
+///     AaCompromised = 10
+/// }
+/// ```
+///
+/// Note that the derive macro will write a `TryFrom<...>` impl for the
+/// provided `#[repr]`, which is used by the decoder.
+#[proc_macro_derive(Integer)]
+#[proc_macro_error]
+pub fn derive_integer(input: TokenStream) -> TokenStream {
+    let input = parse_macro_input!(input as DeriveInput);
+    DeriveEnumerated::new(input, true).to_tokens().into()
 }
```

Comment:
This seems like you're somewhat needlessly defining another proc macro which is ultimately just a 1-argument wrapper for `Enumerated`.

`Integer` is really not very descriptive and kind of ambiguous here: what it's doing is mapping integers to enums, just like `Enumerated`.

I'd suggest just making the `Enumerated` macro more flexible instead.

Reply:
Add I said before, I already tried that. But Rust doesn't remove the type-level attribute, so it is passed through to the final AST and generates a compile error.

Reply:
That must've been a bug in your code, as I've written many proc macros that do that without issue, including this one (see `TypeAttrs`)